From c66a981ffe932a15b5d0ca5586d743792f7b6c0a Mon Sep 17 00:00:00 2001
Description: Get the source code for functions defined in startup.m2.in.
 This file only exists in the Macaulay2 source code and is not installed.
 However, its contents are available as a string, and we can still
 access it.  The code to do this exists in code.m2, but was still
 set up for when these functions were defined in layout.m2, which hasn't
 existed since fb2a04e.
Origin: https://github.com/Macaulay2/M2/pull/1178/commits/c66a981

--- a/M2/Macaulay2/m2/code.m2
+++ b/M2/Macaulay2/m2/code.m2
@@ -8,7 +8,7 @@
      (filename,start,startcol,stop,stopcol,pos,poscol) -> if filename =!= "stdio" then (
 	  wp := set characters " \t\r);";
 	  file := (
-	       if filename === "layout.m2" then startupString
+	       if match("startup.m2.in$", filename) then startupString
 	       else if filename === "currentString" then currentString
 	       else (
 		    if not fileExists filename then error ("couldn't find file ", filename);
